config/environments/test.rb

```diff
@@ -41,6 +41,7 @@
   # The :test delivery method accumulates sent emails in the
   # ActionMailer::Base.deliveries array.
   config.action_mailer.delivery_method = :test
+  config.action_mailer.default_url_options = { host: 'localhost', port: 3001 }
```

Comment:
A minor thing. Why is this 3001?

Reply:
Because I copied it directly from the environment.rb when I removed it and made environment specific ones, wheeeee 😎 removed.